metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -62,6 +63,13 @@ ::map_to_nodes_sequentially(std::string_view sequence,
     std::vector<node_index> path;
     path.reserve(sequence.size() - get_k() + 1);
 
+    if (const auto sshash = std::dynamic_pointer_cast<const DBGSSHash>(graph_)) {
```

Comment:
Why do we need to do it here, and why only for DBGSSHash?

Reply:
DBGSSHash and DBGSuccinct have special functionality that allows us to implement these methods more efficiently, whereas the other types work fine with the naive implementation

Reply:
I don't really understand... Why do we continue mapping here, even where there is a mismatch? It seems inconsistent with how other graph types are handled. And I don't really see how the implementation is more efficient this way? I mean, `map_to_nodes_sequentially` does exactly this when `mode == CANONICAL` anyway?

Reply:
The idea here is that if a k-mer's not found (a mismatch), we don't know if the k-mer is truly missing, or if it's only present in the reverse complement orientation. So, we have to check both. With the SSHash graph, the underlying dictionary class already handles this check, so I'm delegating this logic to that class (assuming that it's more efficient than us implementing the logic with its API)

Reply:
Okay, fair enough I suppose.